homeassistant/helpers/check_config.py

```diff
@@ -181,7 +181,15 @@ def _comp_error(ex: Exception, domain: str, config: ConfigType) -> None:
         if config_schema is not None:
             try:
                 config = config_schema(config)
-                result[domain] = config[domain]
+                if domain in config:
+                    result[domain] = config[domain]
+                else:
+                    result.add_error(
+                        f"Config found for {domain} integration in "
+                        "configuration.yaml. YAML support for this "
+                        "integration has been removed. Please remove "
+                        "it from your configuration.yaml."
+                    )
```

Comment:
I don't really see the point of adding this here, this should be the job of the cv.removed validator.
```suggestion
```